tilavarauspalvelu/typing.py

```diff
@@ -40,7 +40,7 @@ class permission(classmethod): ...  # noqa: N801
 type Lang = Literal["fi", "sv", "en"]
 type TextSearchLang = Literal["finnish", "english", "swedish"]
 type M2MAction = Literal["pre_add", "post_add", "pre_remove", "post_remove", "pre_clear", "post_clear"]
-type EmailContext = dict[str, str | None]
+type EmailContext = dict[str, str | int | datetime.datetime | Decimal | None]
```

Comment:
I wouldn't allow datetime in the email context, since it should probably be formatted in the local datetime-format before applying to the template. By default, datetimes would show up as ISO strings, which is not the best for customer-facing data.

Reply:
Not sure about Decimals, likely you'd want to make sure that it's formatted for displaying prices, although you could do this in the template aswell.

Reply:
I think i added `datetime.datetime` there because I saw it being returned somewhere, but im not finding that anymore... 🤔 
Price we use only for Prices (I think) and they are already being formatted in the template.